web3.js/src/connection.ts

```diff
@@ -3878,7 +3878,14 @@ export class Connection {
   ): Promise<RpcResponseAndContext<SimulatedTransactionResponse>> {
     let transaction;
     if (transactionOrMessage instanceof Transaction) {
-      transaction = transactionOrMessage;
+      let originalTx: Transaction = transactionOrMessage;
+      transaction = new Transaction({
+        recentBlockhash: originalTx.recentBlockhash,
+        nonceInfo: originalTx.nonceInfo,
```

Comment:
You might like to copy this too, since it's mutable. I want to go all over the code and add `Readonly<...>` to every object type, so that we can enforce immutability with TypeScript instead of worrying!